pkg/descheduler/strategies/topologyspreadconstraint.go

```diff
@@ -311,7 +317,21 @@ func balanceDomains(
 func nodesPodFitsOnBesidesCurrent(pod *v1.Pod, nodeMap map[string]*v1.Node) int {
 	count := 0
 	for _, node := range nodeMap {
-		if nodeutil.PodFitsCurrentNode(pod, node) && node != nodeMap[pod.Spec.NodeName] {
+		if node != nodeMap[pod.Spec.NodeName] && nodeutil.PodFitsCurrentNode(pod, node) {
+			count++
+		}
+	}
+	return count
+}
+
+// nodesPodToleratedOnBesidesCurrent counts the number of nodes this pod could fit on based on its toleration and the node's taints.
+func nodesPodToleratedOnBesidesCurrent(pod *v1.Pod, nodeMap map[string]*v1.Node) int {
```

Comment:
does this need to be a new function or could it just be an additional check in `nodesPodFitsOnBesidesCurrent`?

Reply:
The intention was to have a more clear warning message for debugging purposes. `"Ignoring pod for eviction due to node taints not being tolerated"`
Happy to change if you suggest.

Reply:
Yeah, I think combining these is better than looping over the `nodeMap` twice. You could update `nodesPodFitsOnBesidesCurrent` to return an error about what check specifically failed

Reply:
Note that `nodesPodFitsOnBesidesCurrent` check only runs when there is a nodeSelector affinity condition required during scheduling.
This is not the case for taints.

Reply:
Ah, good point. Though, we could move that conditional inside the function and skip the selector/affinity check then. Wdyt?

Reply:
I like it. Done, look forward to your feedback.